climada/entity/measures/base.py

```diff
@@ -272,7 +272,9 @@ def _calc_impact(self, new_exp, new_impfs, new_haz, assign_centroids, save_mat=F
         from climada.engine.impact_calc import ImpactCalc  # pylint: disable=import-outside-toplevel
         imp = ImpactCalc(new_exp, new_impfs, new_haz)\
               .impact(save_mat=save_mat, assign_centroids=assign_centroids)
-        return imp.calc_risk_transfer(self.risk_transf_attach, self.risk_transf_cover)
+        if self.risk_transf_attach==0 and self.risk_transf_cover==0:
```

Comment:
```suggestion
        if self.risk_transf_attach!=0 and self.risk_transf_cover!=0:
```
Is it not that what you want?

Reply:
Also, can you please a test for that?

Reply:
i'll add the tests tomorrow